src/tensorinfo.cpp

```diff
@@ -232,13 +233,49 @@ std::shared_ptr<TensorInfo> TensorInfo::createCopyWithEffectiveDimensionPrefix(c
     return copy;
 }
 
+std::shared_ptr<TensorInfo> TensorInfo::createIntersection(const TensorInfo& other) {
+    if (this->isTensorUnspecified())
+        return std::make_shared<TensorInfo>(other);
+    if (other.isTensorUnspecified())
+        return std::make_shared<TensorInfo>(*this);
+    if ((this->getName() != other.getName()) ||
```

Comment:
What if the tensor names are different because of aliasing in DAG? Aliases can make the output names fit to each other by another level of mapping - aliases.